File: ruby/lib/svix/models/connector_in.rb
# frozen_string_literal: true
# This file is @generated
require "json"

module Svix
  class ConnectorIn
    attr_accessor :name
    # The Connector's UID.
    attr_accessor :uid
    attr_accessor :logo
    attr_accessor :description
    attr_accessor :kind
    attr_accessor :instructions
    attr_accessor :allowed_event_types
    attr_accessor :transformation
    attr_accessor :feature_flags
    attr_accessor :product_type

    ALL_FIELD ||= [
      "name",
      "uid",
      "logo",
      "description",
      "kind",
      "instructions",
      "allowed_event_types",
      "transformation",
      "feature_flags",
      "product_type"
    ].freeze
    private_constant :ALL_FIELD

    def initialize(attributes = {})
      unless attributes.is_a?(Hash)
        fail(ArgumentError, "The input argument (attributes) must be a hash in `Svix::ConnectorIn` new method")
      end

      attributes.each do |k, v|
        unless ALL_FIELD.include?(k.to_s)
          fail(ArgumentError, "The field #{k} is not part of Svix::ConnectorIn")
        end

        instance_variable_set("@#{k}", v)
        instance_variable_set("@__#{k}_is_defined", true)
      end
    end

    def self.deserialize(attributes = {})
      attributes = attributes.transform_keys(&:to_s)
      attrs = Hash.new
      attrs["name"] = attributes["name"]
      attrs["uid"] = attributes["uid"]
      attrs["logo"] = attributes["logo"]
      attrs["description"] = attributes["description"]
      attrs["kind"] = Svix::ConnectorKind.deserialize(attributes["kind"]) if attributes["kind"]
      attrs["instructions"] = attributes["instructions"]
      attrs["allowed_event_types"] = attributes["allowedEventTypes"]
      attrs["transformation"] = attributes["transformation"]
      attrs["feature_flags"] = attributes["featureFlags"]
      if attributes["productType"]
        attrs["product_type"] = Svix::ConnectorProduct.deserialize(attributes["productType"])
      end

      new(attrs)
    end

    def serialize
      out = Hash.new
      out["name"] = Svix::serialize_primitive(@name) if @name
      out["uid"] = Svix::serialize_primitive(@uid) if @uid
      out["logo"] = Svix::serialize_primitive(@logo) if @logo
      out["description"] = Svix::serialize_primitive(@description) if @description
      out["kind"] = Svix::serialize_schema_ref(@kind) if @kind
      out["instructions"] = Svix::serialize_primitive(@instructions) if @instructions
      out["allowedEventTypes"] = Svix::serialize_primitive(@allowed_event_types) if @allowed_event_types
      out["transformation"] = Svix::serialize_primitive(@transformation) if @transformation
      out["featureFlags"] = Svix::serialize_primitive(@feature_flags) if @feature_flags
      out["productType"] = Svix::serialize_schema_ref(@product_type) if @product_type
      out
    end

    # Serializes the object to a json string
    # @return String
    def to_json
      JSON.dump(serialize)
    end
  end
end

File: csharp/Svix/Models/ConnectorIn.cs
// this file is @generated
using System.Text;
using Newtonsoft.Json;

namespace Svix.Models
{
    public class ConnectorIn
    {
        [JsonProperty("allowedEventTypes")]
        public List<string>? AllowedEventTypes { get; set; } = null;

        [JsonProperty("description")]
        public string? Description { get; set; } = null;

        [JsonProperty("featureFlags")]
        public List<string>? FeatureFlags { get; set; } = null;

        [JsonProperty("instructions")]
        public string? Instructions { get; set; } = null;

        [JsonProperty("kind")]
        public ConnectorKind? Kind { get; set; } = null;

        [JsonProperty("logo")]
        public string? Logo { get; set; } = null;

        [JsonProperty("name", Required = Required.Always)]
        public required string Name { get; set; }

        [JsonProperty("productType")]
        public ConnectorProduct? ProductType { get; set; } = null;

        [JsonProperty("transformation", Required = Required.Always)]
        public required string Transformation { get; set; }

        [JsonProperty("uid")]
        public string? Uid { get; set; } = null;

        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();

            sb.Append("class ConnectorIn {\n");
            sb.Append("  AllowedEventTypes: ").Append(AllowedEventTypes).Append('\n');
            sb.Append("  Description: ").Append(Description).Append('\n');
            sb.Append("  FeatureFlags: ").Append(FeatureFlags).Append('\n');
            sb.Append("  Instructions: ").Append(Instructions).Append('\n');
            sb.Append("  Kind: ").Append(Kind).Append('\n');
            sb.Append("  Logo: ").Append(Logo).Append('\n');
            sb.Append("  Name: ").Append(Name).Append('\n');
            sb.Append("  ProductType: ").Append(ProductType).Append('\n');
            sb.Append("  Transformation: ").Append(Transformation).Append('\n');
            sb.Append("  Uid: ").Append(Uid).Append('\n');
            sb.Append("}\n");
            return sb.ToString();
        }
    }
}
